sdk/python/kfp/local/placeholder_utils.py

```diff
@@ -347,6 +350,10 @@ def resolve_individual_placeholder(
             pipeline_task_id,
         dsl.PIPELINE_ROOT_PLACEHOLDER:
             pipeline_root,
+        dsl.WORKSPACE_PATH_PLACEHOLDER:
+            config.LocalExecutionConfig.instance.workspace_root
+            if config.LocalExecutionConfig.instance else tempfile.mkdtemp(
```

Comment:
Could you raise an exception if `config.LocalExecutionConfig.instance.workspace_root` isn't set rather than defaulting to a temp dir here? The reason is that we only want the default to be set at the beginning of the local pipeline run. Having it here causes a different temporary directory to be created and used for each pipeline step using the workspace.